packages/@lwc/engine-core/package.json

```diff
@@ -46,6 +46,7 @@
         "@lwc/shared": "6.0.0"
     },
     "devDependencies": {
-        "observable-membrane": "2.0.0"
+        "observable-membrane": "2.0.0",
+        "@lwc/signals": "6.0.0"
```

Comment:
This should be a dependency, or else it's gonna break things! (Alternatively, make the imports in `mutation-tracker` / `signal-tracker` conditional.)

Reply:
@wjhsf The way `devDependencies` work in this repo is that they get inlined into the built `dist` files. So we use this as a kind of hack when we want certain built files not to contain any `require`s/`import`s.

In this case, I think this is the right approach since historically `@lwc/engine-core` doesn't have any imports. The question then is why not just make `@lwc/signals` part of `@lwc/engine-core` itself.

The answer to that, I think, is that `@lwc/signals` kind of acts as a documentation hub and an ergonomic way to import a base class that consumers can extend. The value of this does seem debatable to me (we could just export the same thing from `@lwc/engine-dom`/`lwc`), and historically we've annoyed users with a proliferation of `@lwc/*` repos they've had to install, and plus this is an experimental feature, so I would lean towards just moving it back to `@lwc/engine-core` for now.